src/Transformer.ts

```diff
@@ -0,0 +1,400 @@
+import * as path from "path";
+import * as ts from "typescript";
+
+import {CompilerOptions} from "./CompilerOptions";
+import {DecoratorKind} from "./Decorator";
+import {TSTLErrors} from "./Errors";
+import {TransformHelper as transformHelper} from "./TransformHelper";
+import {TSHelper as tsHelper} from "./TSHelper";
+
+export class LuaTransformer {
+    private checker: ts.TypeChecker;
+    private options: CompilerOptions;
+    private context: ts.TransformationContext;
+
+    constructor(checker: ts.TypeChecker, options: CompilerOptions) {
+        this.checker = checker;
```

Comment:
```ts
constructor(
    private checker: ts.TypeChecker,
    private context: ts.TransformationContext,
    private options: CompilerOptions
) { }
```
Is slightly nicer imo

Reply:
I think this is only nicer if all members are declared & initialized in the constructor, which is not the case here.